### DISCUSSION TITLE
Exploring LXQt's Lightweight Desktop Philosophy and Getting Started with Customization

### DISCUSSION THREAD

**Discussion by kalgnana** [lxqt/lxqt#2492] [Q&A]
Hi LXQt Team,

I have over 15 years of experience in system programming with a focus on embedded application development. My expertise lies in the C programming language.

Recently, we were running Ubuntu (using the GTK toolkit) on a less powerful machine with 4GB of RAM and a dual-core processor. Unfortunately, we experienced significant slowness and noticeable latency in user interactions.

At the same time, when we switched to Lubuntu (using the QT toolkit) on the same machine, we did not encounter any latency issues. However, we did find that the look and feel of Lubuntu, particularly the network manager, was not as visually appealing as Ubuntu.

I understand that LXQt's development philosophy is centered around providing a lightweight, fast, and efficient desktop environment. We are interested in delving deeper into how this philosophy is translated into the implementation and design of the LXQt repository architecture. We would like to customise LXQt to meet our needs and contribute to the community.

However, I'm unsure where to begin and how to start this process. Could someone please provide guidance on how to get started and point me towards relevant documentation for understanding the existing design?

Regards,
Kalyan-G.

**Comment by stefonarch** [comment]
One main point is: LXQt provides only those applications and tools which you can find in our repos. The choice of the WM or networkmanager (I'm using CMST)  and other components is done by distros (or users).

**Reply by tsujan** [reply]
> ... was not as visually appealing as....

How apps look depends on the icon set and the style their toolkit uses, among other things. Qt apps use the active Qt style. Qt's default style is called Fusion, but there are other styles — with different looks and feels, from simple to complex, and with various customization possibilities — which you could use. Also, there are many icon themes.

However, LXQt itself doesn't provide or enforce any Qt style or icon set; it is (and will be) neutral toward them. The only styles LXQt provides (without forcing them) are "LXQt styles", which are mainly used by its own Panel and Runner, not by other Qt apps.

**Comment by AzumaHazuki** [comment]
I'm your girl for this one :) Have been doing lots of ports of old-school themes to LXQt's style system, mostly the GTK2 ones. My best work is probably a port of the _entirety_ of the old xfce-gtk2-engine collection.

What kind of look are you after? I can't guarantee anything up front but might be able to make something for you.

**Reply by romildo** [reply]
It would be nice to have themes following the [nord colors and palettes](https://www.nordtheme.com).

**Reply by AzumaHazuki** [reply]
I've actually been working on those...it turns out, _all_ of EliverLara's GTK themes are basically the same aside from the colors! So if I can make a basic one and just use Sed to sub in colors, in theory I can have Nordic, Juno, etc. all done fairly easily
